app/code/Magento/InventoryInStorePickup/Model/Source/SetExtensionAttributes.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\Source;
+
+use Magento\Framework\Api\ExtensionAttributesFactory;
+use Magento\Framework\DataObject;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryInStorePickupApi\Api\Data\PickupLocationInterface;
+
+/**
+ * Set store-pickup related source extension attributes
+ */
+class SetExtensionAttributes
+{
+    /**
+     * @var ExtensionAttributesFactory
+     */
+    private $extensionAttributesFactory;
+
+    /**
+     * @param ExtensionAttributesFactory $extensionAttributesFactory
+     */
+    public function __construct(ExtensionAttributesFactory $extensionAttributesFactory)
+    {
+        $this->extensionAttributesFactory = $extensionAttributesFactory;
+    }
+    /**
+     * @param SourceInterface|DataObject $source
+     */
+    public function execute(SourceInterface $source): void
+    {
+        $pickupAvailable = $source->getData(PickupLocationInterface::IS_PICKUP_LOCATION_ACTIVE);
+        $frontendName = $source->getData(PickupLocationInterface::FRONTEND_NAME);
+        $frontendDescription = $source->getData(PickupLocationInterface::FRONTEND_DESCRIPTION);
```

Comment:
These fields doesn't seem to be present in Pickup Location entity.

Reply:
Yes, they are not: they are extension attribute codes of `Source`.

I am not sure there is a sense to move these constants elsewhere, - to a separate class with just constants only.

Or it is?